spec/controllers/vm_cloud_controller_spec.rb

```diff
@@ -18,64 +18,50 @@
       ApplicationController::Explorer::X_BUTTON_ALLOWED_ACTIONS.each_pair do |action_name, method|
         prefixes = ["image", "instance"]
         prefixes.each do |prefix|
-          actual_action = "#{prefix}_" + action_name
-          actual_method = if method == :s1 || method == :s2
-                            "#{prefix}_" + action_name
-                          else
-                            method.to_s
-                          end
+          actual_action = "#{prefix}_#{action_name}"
+          actual_method = [:s1, :s2].include?(method) ? actual_action : method.to_s
+
           it "calls the appropriate method: '#{actual_method}' for action '#{actual_action}'" do
             controller.stub(:x_button_response)
             controller.should_receive(actual_method)
-            get :x_button, :id => FactoryGirl.create(:template_redhat), :pressed => actual_action
+            get :x_button, :id => nil, :pressed => actual_action
           end
         end
       end
     end
 
     it 'exception is raised for unknown action' do
-      get :x_button, :id => FactoryGirl.create(:template_redhat), :pressed => 'random_dude', :format => :html
+      get :x_button, :id => nil, :pressed => 'random_dude', :format => :html
```

Comment:
Are you sure it's the *right* exception and not just something like "can't find id nil"?